python/src/scipp/show.py

```diff
@@ -380,14 +380,25 @@ def make_svg(self):
         for what, items in categories:
             for name, var in items.items():
                 item = DrawerItem(name, var, config.colors[what])
-                if len(var.dims) == 0:
-                    area_0d.append(item)
-                elif var.dims[-1] == dims[-1]:
-                    area_x.append(item)
-                elif var.dims[-1] == dims[-2]:
-                    area_y.append(item)
-                else:
-                    area_z.append(item)
+                try:
+                    if len(var.dims) == 0:
+                        area_0d.append(item)
+                    elif var.dims[-1] == dims[-1]:
+                        area_x.append(item)
+                    elif var.dims[-1] == dims[-2]:
+                        area_y.append(item)
+                    elif var.dims[-1] == dims[-3]:
+                        area_z.append(item)
+                    else:
+                        # dirty hack to trigger error handling below
+                        raise IndexError()
```

Comment:
I'm not sure I understand why you need the `try` block with the `raise IndexError` and not just have
```Py
else:
    raise sc.DimensionError(
                        "Cannot show DataArray, it contains a coordinate with"
                        f" a dimension label ('{var.dims[-1]}') that is not "
                        "contained in the data") from None
```

Reply:
If the data array is e.g. 2 dimensional, the last `elif` will throw. The `try``-except` is just a way of not writing `if len(dims) > N and ...` in every branch.